src/region.c

```diff
@@ -182,7 +182,7 @@ gdip_is_InfiniteRegion (GpRegion *region)
 		/* FIXME: incomplete and not 100% accurate (curves) - but cover the most common case */
 		if (region->tree && region->tree->path && (region->tree->path->count == 4)) {
 			GpRectF bounds;
-			if (GdipGetPathWorldBounds (region->tree->path, &bounds, NULL, NULL) != Ok)
+			if (GdipGetPathWorldBounds (region->tree->path, &bounds, NULL, NULL) == Ok)
```

Comment:
Weirdly enough, this seems to trip one of the unit tests (creating a path, adding a curve with REGION_INFINITE_POSITION/REGION_INFINITE_LENGTH coordinates and checking if a region created from the path is infinite).